plugin/plugins.go

```diff
@@ -0,0 +1,155 @@
+package plugin
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/alibaba/sealer/common"
+	"github.com/alibaba/sealer/logger"
+	v1 "github.com/alibaba/sealer/types/api/v1"
+	"github.com/alibaba/sealer/utils"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+/*
+config in PluginConfig:
+
+apiVersion: sealer.aliyun.com/v1alpha1
+kind: Plugin
+metadata:
+  name: SHELL
+spec:
+  action: PostInstall
+  on: role=master
+  data: |
+    kubectl taint nodes node-role.kubernetes.io/master=:NoSchedule
+
+Dump will dump the config to etc/redis-config.yaml file
+*/
+
+type Plugins interface {
+	Dump(clusterfile string) error
+	Run(cluster *v1.Cluster, phase Phase) error
+}
+
+type DumperPlugin struct {
+	configs     []v1.Plugin
+	clusterName string
+}
+
+func (c *DumperPlugin) GetPhasePlugin(phase Phase) []v1.Plugin {
```

Comment:
you should get all the plugins, not the phase plugin. every plugin will decide to run it or not. some plugin config the `spec.on` field is nil.

Reply:
I want the plug-in to execute fast enough, so I use the smallest number I can get because of the identity, to make the execution more efficient